src/apps/basket/tests.py

```diff
@@ -0,0 +1,81 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from src.apps.basket.basket import Basket
+from src.apps.inventory.models import Product, Category
+
+
+class BasketAddViewTest(TestCase):
+    def setUp(self):
+        self.category = Category.objects.create(name="Test Category")
+        self.user_seller = get_user_model().objects.create_user(
+            username="seller", password="password", user_type="Seller"
+        )
+        self.product = Product.objects.create(
+            name="Test Product",
+            category=self.category,
+            price=10.0,
+            seller=self.user_seller,
+            stock=10,
+        )
+
+    def test_add_product_to_basket(self):
+        self.client.login(username="testuser", password="testpassword")
+        basket = Basket(self.client)
+        initial_quantity = basket.get_quantity(self.product)
+        response = self.client.post(
+            reverse("basket:basket_add", args=[self.product.id]),
+            {"quantity": 5, "update": False},
+            HTTP_REFERER="/",
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Basket(self.client).get_quantity(self.product), initial_quantity + 5)
+
+    def test_add_product_to_basket_insufficient_stock(self):
+        self.client.login(username="testuser", password="testpassword")
+        basket = Basket(self.client)
+        initial_quantity = basket.get_quantity(self.product)
+        response = self.client.post(
+            reverse("basket:basket_add", args=[self.product.id]),
+            {"quantity": 11, "update": False},
+            HTTP_REFERER="/",
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Basket(self.client).get_quantity(self.product), initial_quantity)
+
```

Comment:
do user get any error message? 
if yes, we can test it as well

Reply:
Yes, I can add such functionality